gnucash/import-export/aqb/dialog-ab-select-imexporter.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * dialog-ab-select-imexporter.c --
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, contact:
+ *
+ * Free Software Foundation           Voice:  +1-617-542-5942
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org
+ */
+
+/**
+ * @internal
+ * @file dialog-ab-select-imexporter.h
+ * @brief Dialog to select AQBanking importer/exporter and format profile.
+ * @author  Copyright (C) 2022 John Ralls <jralls@ceridwen.us>
+ */
+
+#include <config.h>
+
+#include <glib/gi18n.h>
+#include "dialog-ab-select-imexporter.h"
+#include <dialog-utils.h>
+
+static QofLogModule log_module = G_LOG_DOMAIN;
+
+struct _GncABSelectImExDlg
+{
+    GtkWidget *dialog;
+    GtkWidget *parent;
+    GtkListStore *imexporter_list;
+    GtkListStore *profile_list;
+    GtkWidget *select_imexporter;
+    GtkWidget *select_profile;
+    GtkWidget *ok_button;
+
+    AB_BANKING* abi;
+};
+
+// Expose the selection handlers to GtkBuilder.
+G_MODULE_EXPORT gboolean imexporter_changed(GtkTreeSelection* sel,
+                                            gpointer data);
+G_MODULE_EXPORT gboolean profile_changed(GtkTreeSelection* sel, gpointer data);
+
+static gboolean
+clear_model_row (GtkTreeModel* model, GtkTreePath* path,
+                 GtkTreeIter* iter, gpointer data)
+{
+    char *name, *desc;
+    g_return_val_if_fail (model && iter, TRUE);
+    gtk_tree_model_get (model, iter, 0, &name, 1, &desc, -1);
```

Comment:
I know there are only two columns in model but I would create an enum so names can be used instead of numbers.

Reply:
Even better: I doubt that this whole `clear_model_row` is needed in the first place -- its purpose is to `g_free` the model data, however, `gtk_list_store_clear` will take care of it. AFAIU The `g_free` here is simply freeing the copies provided by `gtk_tree_model_get`.

Reply:
You're both right.